package/network/utils/iproute2/Makefile

```diff
@@ -19,9 +19,8 @@ PKG_BUILD_DEPENDS:=iptables
 PKG_LICENSE:=GPL-2.0
 PKG_CPE_ID:=cpe:/a:iproute2_project:iproute2
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
-
 include $(INCLUDE_DIR)/kernel.mk
+undefine PKG_BUILD_DIR
```

Comment:
curios : why are these `undefine PKG_BUILD_DIR` needed ?

Reply:
[to add here] my sentiment, is that they look hacky;

Reply:
IIRC because kernel.mk sets a `PKG_BUILD_DIR` that is wrong (I think kernel.mk is meant more for kernel modules)

Reply:
hmm; these are very subtle changes here;
it looks to me that these are changes that handle differences between SDK generation, imagebuilder & full target build (so, 3 build-modes) ;

i would first find a way to remove this elegantly from `kernel.mk`:
```
  ifneq ($(TARGET_BUILD),1)
    PKG_BUILD_DIR ?= $(KERNEL_BUILD_DIR)/$(PKG_NAME)$(if $(PKG_VERSION),-$(PKG_VERSION))
  endif
```

once that's done, this patch be straightforward;

Reply:
Honestly, I don't understand the build system enough to be confident of a change like that :joy:

The original iproute2 (and ppp) Makefile already overrides `PKG_BUILD_DIR`; to undefine it here and let package.mk set it to the default value is the smallest (and safest) change I can make.